Add - Anonymous Chat

gokulkrishh opened the pull request:
**Contain**

1. Manifest.json
2. Service Worker
3. Add to home screen
4. Served via https
5. Static assets are cached

**Site:** https://anonymouschat.in

pepelsbey commented:
Thank you for suggestion! But all I see so far is a spinner

![image](https://cloud.githubusercontent.com/assets/105274/17694314/ec74ee54-63ab-11e6-9ed9-1587f960916b.png)

How this suppose to work? Imagine someone browsing this list looking for a good example of PWA, what would this person think when they see an endless spinner and no explanation whatsoever?

pepelsbey commented:
Could you please also update your fork with the latest changes from upstream so `qrcode-scanner` won’t be presented in your current PR? It’s already merged.

gokulkrishh commented:
@pepelsbey Hey, I don't know why you are seeing only the spinner. I have tested the app in chrome, opera, safari also in Android and iOS. Not sure why its not working for you. Can you tell me the which browser and its version you are using.

B/W I have done the fetch from upstream. Not sure why i got some previous commits. I will do the fetch again and update.

pepelsbey commented:
Ok, now I finally see the chat but it takes up to a minute (or sometimes more) or looking at spinner before it goes live. Also input is hidden in Safari so it’s not clear what to do here (see below). It’s also not clear how this thing works: how am I supposed to invite someone here, is it just a single chat room for everyone? A little question mark icon with explanation would make it much better.

![img_6395](https://cloud.githubusercontent.com/assets/105274/17695174/0c900b84-63b0-11e6-8b26-39ca9edadb5e.jpg)

pepelsbey commented:
It’s slow in all browsers I used (Safari, Opera, Chrome, mobile and desktop), so it’s probably related to my location: St. Petersburg, Russia.

gokulkrishh commented:
@pepelsbey It's a 1 on 1 chat application which connects with users randomly. I will add info in intro screen or similar to how you are suggesting. 

Also, app size is **787 KB** and attaching the devTool network screen. Might be due to slow connection or something.

<img width="880" alt="screen shot 2016-08-16 at 3 26 09 pm" src="https://cloud.githubusercontent.com/assets/2944237/17695429/2bbc9a52-63c6-11e6-85c6-0f7b1a4d7fb5.png">

pepelsbey commented:
> It's a 1 on 1 chat application which connects with users randomly

It wasn’t clear at all, worth adding.

gokulkrishh commented:
@pepelsbey Done. I have added a popup to what is this app about. 

![www google co in-_-chrome-newtab-espv 2 ie utf-8 iphone 5](https://cloud.githubusercontent.com/assets/2944237/17843699/bb654ec4-684e-11e6-88d3-4409f28849a0.png)

Also removed old commits and rebased with master.
